File: 0001-two-sum/0001-two-sum.ts
function twoSum(nums: number[], target: number): number[] {
    const map = {}
    
    for (let i=0;i<nums.length;i++){
        const num = nums[i];
        if( map[target-num] !==undefined){
            return [map[target-num],i]
        }
        else {
            map[num] = i
        }
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        complementMap = dict()
        l = len(nums)
        for i in range(l):
            num = nums[i]
            complement = target - num
            if complement in complementMap:
                return [complementMap[complement], i]
            else:
                complementMap[num] = i